detectable/src/main/java/com/synopsys/integration/detectable/detectables/npm/lockfile/NpmDependencyConverter.java

```diff
@@ -118,15 +117,30 @@ public void linkPackagesDependencies(PackageLock packageLock) {
             return;
         }
                 
-        for(String packageName : packageLock.packages.keySet()) { 
```

Comment:
I think we'll be okay here...or at least be no worse off than we were before. That gson.fromJson line has been in place before the recent npm changes. This code is all in the npm lockfile detector and before that detector is chosen we make sure there is a package-lock.json file so that's something. If lockFileText isn't in a good state we'll not really be able to do anything so there is some ground for validating the file before we parse it too much but I'm thinking maybe that's something outside of this PR if we want to do it.

Reply:
We could potentially do validation in NpmPackageLockDetectable.applicable where we current validate the existence of the file. We could possibly add content validation here. I'm not sure if other detectors do this though.